### ISSUE TITLE
Add open and close animations for MenuAnchor

### ISSUE THREAD

**Issue by SeanIsTyping** [flutter/flutter#135025]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

So I noticed this after replacing PopupMenuButtons and DropdownButtons in our app with MenuAnchors and DropdownMenus. 

According to the [material design guidelines](https://m3.material.io/components/menus/guidelines#:~:text=Menus%20use%20a%20fade%20transition%20pattern%20to%20enter%20and%20exit%20the%20screen.%20The%20animation%20creates%20a%20relationship%20between%20the%20menu%20and%20the%20interactive%20element%20that%20generates%20the%20menu.), there should be a fade transition when menu enters and exits on screen, but currently MenuAnchor just opens and close sthe menu without any animations.

### Proposal

It's not a big deal, and would actually be preferable if on desktop and web, but on Android it kinds of look weird for the menus to just suddenly appears.

**Comment by HansMuller** [comment]
Seems like a good idea. It would be good to establish a pattern for these kinds of animation customizations.

@gspencergoog - what do you think?

**Comment by OptimisticSean** [comment]
> @OptimisticSean could you please attach a minimal reproducible code? and btw are you referring to [this](https://m3.material.io/components/menus/guidelines#34ff1f8d-39cc-4fd1-aa63-47b57c385274)?

Yes. For reproducible code I'll just link their API pages ([1](https://api.flutter.dev/flutter/material/DropdownButton-class.html), [2](https://api.flutter.dev/flutter/material/PopupMenuButton-class.html)). As you can see the older widgets both shows an animation when the menu is opened and closed.

**Comment by gspencergoog** [comment]
Yes, it would be good to add some hooks for animating the menus.  There might be some timing issues to work out when we do that (e.g. when to make things traversable or not, when to fire the "selected" callback, etc.), but I think they're solvable.

**Comment by crystalarcus** [comment]
Any progress ? its already June 2024 guys, come this is a component mentioned in official documentation, it should be taken seriously.

**Comment by deminearchiver** [comment]
> Any progress ? its already June 2024 guys, come this is a component mentioned in official documentation, it should be taken seriously.

There is currently a pull request open.

#143416

**Comment by shubhamsinghshubham777** [comment]
> > Any progress ? its already June 2024 guys, come this is a component mentioned in official documentation, it should be taken seriously.
> 
> There is currently a pull request open.
> 
> [#143416](https://github.com/flutter/flutter/pull/143416)

This PR has been closed. Is work on this ticket still in active state?

**Comment by davidhicks980** [comment]
As @deminearchiver pointed out, it is nearing completing. There were some challenges related to ensuring other menus (CupertinoMenuAnchor, in particular) could use the same base widget as MenuAnchor. 

@marcglasberg We are currently only providing an "animated" boolean to limit the scope of the initial PR. However, I expect future PRs will add additional customization, although I'm not sure what shape that customization will take.
